Pseudocode/Dynamic Programming/Bursting Balloons/Source Code.tex

```diff
@@ -0,0 +1,71 @@
+% Set the Page Layout
+\documentclass[12pt]{article}
+\usepackage[inner = 2.0cm, outer = 2.0cm, top = 2.0cm, bottom = 2.0cm]{geometry}
+
+
+% Package to write pseudo-codes
+\usepackage{algorithm}
+
+% Remove the 'end' at the end of the algorithm
+\usepackage[noend]{algpseudocode}
+
+% Define Left Justified Comments
+\algnewcommand{\LeftComment}[1]{\Statex \(\triangleright\) #1}
+
+% Remove the Numbering of the Algorithm
+\usepackage{caption}
+\DeclareCaptionLabelFormat{algnonumber}{Algorithm}
+\captionsetup[algorithm]{labelformat = algnonumber}
+
+\begin{document}
+
+\begin{algorithm}
+
+  \caption{Find the maximum amount of money that can be collected by bursting balloons}
+\begin{algorithmic}[1]
+    \Ensure One Based Indexing for the array
+    \Statex
+ 
+     
+    \Function{$Collect$}{$arr,low,high$}
+    \State $len \gets high-low+1$
+  \If{$len<=2$}
+    
+ 	 	\Return $0$
+   \EndIf
+   \If{$arr[low][high]!=-1$} \Comment denotes the profit by collecting in the range [low,high]
```

Comment:
Serious bug. We are not storing the answer in the array . We are using a memo table.